src/common/admin_socket.cc

```diff
@@ -484,6 +484,25 @@ int AdminSocket::unregister_command(std::string_view command)
   return ret;
 }
 
+void AdminSocket::unregister_commands(const AdminSocketHook *hook)
+{
+  std::unique_lock l(lock);
```

Comment:
nit, `std::scoped_lock` would suffice here.

Reply:
std::scoped_lock is the same std::lock_guard.  There is a wait to need this lock.